libevmasm/RuleList.h

```diff
@@ -99,6 +99,8 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart1(
 		}},
 		{Builtins::ADDMOD(A, B, C), [=]{ return C.d() == 0 ? 0 : Word((bigint(A.d()) + bigint(B.d())) % C.d()); }},
 		{Builtins::MULMOD(A, B, C), [=]{ return C.d() == 0 ? 0 : Word((bigint(A.d()) * bigint(B.d())) % C.d()); }},
+        {Builtins::MOD(ADD(A, B), C), [=]{ return ADDMOD(A, B, C); }},
+        {Builtins::MOD(MUL(A, B), C), [=]{ return MULMOD(A,B,C); }},
```

Comment:
Should be something like
`Builtins::MOD(MUL(X, Y), A), [=]{ return MULMOD(X, Y, A); }`

But you would need to check that `A.d()` is of the form `2**N`.